mono/mini/mini-posix.c

```diff
@@ -246,8 +246,14 @@ MONO_SIG_HANDLER_FUNC (static, sigterm_signal_handler)
 
 	// Will return when the dumping is done, so this thread can continue
 	// running. Returns FALSE on unrecoverable error.
-	if (!mono_threads_summarize_execute (&mctx, &output, &hashes, FALSE, NULL, 0))
-		g_error ("Crash reporter dumper exited due to fatal error.");
+	if (!mono_dump_inprogress ()) {
```

Comment:
This is racy.
I believe you should have a "compare and set" sort of operation instead of "dump_inprogress" and "dump_start": `gboolean mono_dump_try_strart` if it returns TRUE, then a crash dump wasn't in progress and we're the supervisor.  if it returns FALSE then a crash  dump is in progress and we do something else.  In the SIGTERM handler the "something else" is to run the per-thread summary (ie the existing "else" code" you have below).  In the non-SIGTERM signal handlers we will assert that a dump is already in progress.   And in the `DumpStateTotal` case it will sleep and retry.